File: Data Structures & Algorithms/is-palindrome/submission-5.cpp
class Solution {
public:
    bool isPalindrome(string s) {
        int left = 0, right = s.size() - 1;

        while (left < right) {
            if (!isalnum(s[left])) {
                left++;
                continue;
            }

            if (!isalnum(s[right])) {
                right--;
                continue;
            }

            cout << s[left] << " " << s[right] << endl;

            if (tolower(s[left]) != tolower(s[right])) {
                return false;
            }

            left++;
            right--;
        }

        return true;
    }
};

File: Data Structures & Algorithms/three-integer-sum/submission-5.py
class Solution:
    def threeSum(self, nums: List[int]) -> List[List[int]]:
        
        finalAnswer = []
        nums.sort()

        for i,num in enumerate(nums) :
            
            left = i + 1
            right = len(nums) - 1

            while left < right :
                currSum = nums[left] + nums[right] + num 
                if currSum == 0 :
                    if [num,nums[left],nums[right]] not in finalAnswer : 
                        finalAnswer.append([num,nums[left],nums[right]])
                    left += 1
                    right -= 1

                elif currSum > 0 :
                    right -= 1

                elif currSum < 0 :
                    left += 1 

        return finalAnswer
                    